pkgs/development/compilers/chicken/5/chicken.nix

```
@@ -19,43 +19,45 @@ stdenv.mkDerivation rec {
     sha256 = "1yl0hxm9cirgcp8jgxp6vv29lpswfvaw3zfkh6rsj0vkrv44k4c1";
   };
 
-  setupHook = lib.optional (bootstrap-chicken != null) ./setup-hook.sh;
+  nativeBuildInputs = [ makeWrapper ];
+
+  buildInputs = lib.optionals (bootstrap-chicken != null) [
+    bootstrap-chicken
+  ];
 
   # -fno-strict-overflow is not a supported argument in clang on darwin
-  hardeningDisable = lib.optionals stdenv.isDarwin ["strictoverflow"];
+  hardeningDisable = lib.optionals stdenv.isDarwin [ "strictoverflow" ];
 
   makeFlags = [
-    "PLATFORM=${platform}" "PREFIX=$(out)"
+    "PLATFORM=${platform}"
+    "PREFIX=$(out)"
+    "HOSTSYSTEM=${hostPlatform.config}"
+    "TARGETSYSTEM=${targetPlatform.config}"
+    "LIBRARIAN=$(HOSTSYSTEM)-gcc-ar"
+    "C_COMPILER=$(CC)"
+    "CXX_COMPILER=$(CXX)"
   ] ++ (lib.optionals stdenv.isDarwin [
```

```suggestion
  ] ++ lib.optionals stdenv.isDarwin [
```
we don't need those.